services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/credentials/AbstractCredentialsManagementService.java

```diff
@@ -41,23 +43,34 @@
  */
 public abstract class AbstractCredentialsManagementService implements CredentialsManagementService {
 
-    protected TenantInformationService tenantInformationService = new NoopTenantInformationService();
 
-    private HonoPasswordEncoder passwordEncoder;
+    protected TenantInformationService tenantInformationService = new NoopTenantInformationService();
 
     private final Vertx vertx;
+    private final HonoPasswordEncoder passwordEncoder;
+    private final int maxBcryptIteration;
+    private final Set<String> hashAlgorithmsWhitelist;
 
     /**
      * Creates a service for the given Vertx and password encoder instances.
      *
      * @param vertx The Vertx instance to use.
      * @param passwordEncoder The password encoder.
-     * @throws NullPointerException if any of the parameters is {@code null};
+     * @param maxBcryptIteration The maximum number of allowed bcrypt iterations.
+     * @param hashAlgorithmsWhitelist An optional collection of allowed password hashes.
+     * @throws NullPointerException if any of the required parameters is {@code null};
      */
-    @Autowired
-    public AbstractCredentialsManagementService(final Vertx vertx, final HonoPasswordEncoder passwordEncoder) {
+    public AbstractCredentialsManagementService(
+            final Vertx vertx,
+            final HonoPasswordEncoder passwordEncoder,
+            final int maxBcryptIteration,
+            final Set<String> hashAlgorithmsWhitelist) {
+
         this.vertx = Objects.requireNonNull(vertx);
         this.passwordEncoder = Objects.requireNonNull(passwordEncoder);
+        this.maxBcryptIteration = maxBcryptIteration;
+        this.hashAlgorithmsWhitelist = hashAlgorithmsWhitelist;
```

Comment:
Null check is missing for `hashAlgorithmsWhitelist`. `hashAlgorithmsWhitelist` is passed as an argumet to 
 `DeviceRegistryUtils.checkCredential(...)` and it does not allow _null_ value for this argument.